### PR TITLE
Add what’s new page for Firefox 119 [fix #13766]

### LINKED ISSUE #13766: What's New page for Firefox 119
## Description

Add a WNP for Firefox 119, releasing on 24 October, 2023.

DE/FR/GB - 33% of traffic receives a page promoting three different privacy/security products, 33% gets the same trio but with collapsed boxes that expand on hover, and the last 33% sees a promotion for Relay.

EN - 50% of traffic sees a similar-but-different trio of privacy products, and the other 50% sees a promotion for a collection of privacy/security addons.

Copy: https://docs.google.com/document/d/1GxmLRQJ8Bl2eU0EJIHtXEAJpyrs4-elTcByi2VWh2tY/edit
Design NA: https://www.figma.com/file/CZvzHgU9NQMQUm34lomIl5/Firefox-WNP?type=design&node-id=4841-4621
Design EU: https://www.figma.com/file/8fVLbzzAd40lSSLuOwCfnl/EU_Firefox_WNP_21-23?type=design&node-id=3679%3A6

### PR DESCRIPTION
**Pull request by craigcook**
## One-line summary
WNP for Firefox 119, releasing on **24 October, 2023**

Three variations for EU (en-GB, de, fr): a trio of products, the same trio in collapsable boxes, and Relay.
Two variations for NA (en-US, en-CA): a trio of products, and an addons collection.

## Issue / Bugzilla link
#13766

## Testing

NA v1: http://localhost:8000/en-US/firefox/119.0/whatsnew/?v=1
NA v2: http://localhost:8000/en-US/firefox/119.0/whatsnew/?v=2
EU en-GB v1: http://localhost:8000/en-GB/firefox/119.0/whatsnew/?v=1
EU en-GB v2: http://localhost:8000/en-GB/firefox/119.0/whatsnew/?v=2
EU en-GB v3: http://localhost:8000/en-GB/firefox/119.0/whatsnew/?v=3
EU de v1: http://localhost:8000/de/firefox/119.0/whatsnew/?v=1
EU de v2: http://localhost:8000/de/firefox/119.0/whatsnew/?v=2
EU de v3: http://localhost:8000/de/firefox/119.0/whatsnew/?v=3
EU fr v1: http://localhost:8000/fr/firefox/119.0/whatsnew/?v=1
EU fr v2: http://localhost:8000/fr/firefox/119.0/whatsnew/?v=2
EU fr v3: http://localhost:8000/fr/firefox/119.0/whatsnew/?v=3

### CODE REVIEW

File: bedrock/firefox/views.py [hunk 1]
```diff
@@ -515,6 +522,21 @@ def get_template_names(self):
                 template = "firefox/developer/whatsnew.html"
             else:
                 template = "firefox/whatsnew/index.html"
+        elif version.startswith("119."):
+            if locale in ["de", "fr", "en-GB"] or country == "GB":
```
Comment (alexgibson) [inline bedrock/firefox/views.py:526]: Issue: If I'm in the UK, this means I always see this template irrespective of my language?

http://localhost:8000/pl/firefox/119.0/whatsnew/?geo=gb

File: media/css/firefox/whatsnew/whatsnew-119-eu-trio.scss [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+            .c-box-content {
+                height: 0;
+                overflow: hidden;
+                transition: height 350ms ease-in-out;
+            }
+
+            .c-box-inner:hover,
+            .c-box-inner:focus,
+            .c-box-inner:focus-within {
```
Comment (alexgibson) [inline media/css/firefox/whatsnew/whatsnew-119-eu-trio.scss:126]: 🔥

File: media/js/firefox/whatsnew/whatsnew-119-eu-relay.es6.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    document.querySelector('.wnp-content-main').classList.add('static');
+}
+
+(function () {
+    Mozilla.Client.getFxaDetails(function (details) {
+        if (details.setup) {
+            document.querySelector('.wnp-content-main');
+        }
+    });
+})();
```
Comment (alexgibson) [inline media/js/firefox/whatsnew/whatsnew-119-eu-relay.es6.js:47]: Curious - what is this doing? It seems like it might be an incomplete DOM query - is it needed?

File: media/js/firefox/whatsnew/whatsnew-119-experiment-eu.es6.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+const initTrafficCop = () => {
+    if (href.indexOf('v=') !== -1) {
+        if (href.indexOf('v=1') !== -1) {
+            window.dataLayer.push({
+                'data-ex-variant': 'wnp119-boxes-open',
+                'data-ex-name': 'wnp-119-experiment-eu'
+            });
+        } else if (href.indexOf('v=2') !== -1) {
+            window.dataLayer.push({
+                'data-ex-variant': 'wnp117-boxes-closed',
```
Comment (alexgibson) [inline media/js/firefox/whatsnew/whatsnew-119-experiment-eu.es6.js:21]: 119?

```suggestion
                'data-ex-variant': 'wnp119-boxes-closed',
```

**Review by alexgibson — CHANGES_REQUESTED** [review]
Spotted one issue with the template variation logic and a couple of small nits, but otherwise this all looks great. Nice work!

File: bedrock/firefox/views.py [hunk 5]
```diff
@@ -515,6 +522,21 @@ def get_template_names(self):
...
+                if variant == "3":
+                    template = "firefox/whatsnew/whatsnew-fx119-eu-relay.html"
+                elif variant == "2":
+                    template = "firefox/whatsnew/whatsnew-fx119-eu-trio-v2.html"
+                else:
+                    template = "firefox/whatsnew/whatsnew-fx119-eu-trio-v1.html"
```
Comment (alexgibson) [inline bedrock/firefox/views.py:532]: Oh, these two templates also seem to be missing the experiment JS, so the Traffic Cop redirect isn't triggering got the EU versions?

File: media/js/firefox/whatsnew/whatsnew-119-eu-relay.es6.js (hunk 3 above)
Comment (craigcook) [reply media/js/firefox/whatsnew/whatsnew-119-eu-relay.es6.js:47]: Ah good catch. This was a leftover from the same script for WNP 117. Though that page included logic to swap buttons for people who were signed in, but this version doesn't have that (they didn't specify it, and I was afraid to ask because I didn't want to add another complication).

File: bedrock/firefox/views.py (hunk 5 above)
Comment (craigcook) [reply bedrock/firefox/views.py:532]: it's in trio-v1 and trio-v2 just extends that template so it should appear there as well... or do I need to repeat the block in v2?

Comment (alexgibson) [reply bedrock/firefox/views.py:532]: Oh so it is, then how come I didn't get redirected. Hmm, let me try again

Comment (alexgibson) [reply bedrock/firefox/views.py:532]: Ah, never mind - I tested in a private window and that had DNT enabled. Seems to work fine 👍

**Review by alexgibson — APPROVED** [review]
r+
